test/wallet-rpc-test.js

```diff
@@ -0,0 +1,338 @@
+     * 'getaddressinfo', assert that it returns isscript as true and the
+     * correct value as the witness program
+     */
+    it('should detect p2wsh program', async () => {
+      const xpub = testxpub(0, network);
+      const pubkeys = [];
+      pubkeys.push(xpub.derive(0).derive(0).publicKey);
+      pubkeys.push(xpub.derive(0).derive(1).publicKey);
+      const script = Script.fromMultisig(2, 2, pubkeys);
+
+      const address = Address.fromWitnessScripthash(script.sha256());
+
+      const addr = address.toString(network);
+      const response = await wclient.execute('getaddressinfo', [addr]);
+
+      assert.equal(response.isscript, true);
+      assert.equal(response.witness_program, address.hash.toString('hex'));
+    });
+
+    /*
+     * Iterate through the lookahead amount of addresses and request
+     * the rpc 'getaddressinfo' for each, assert that each ismine is
+     * true and then assert that ismine is false when the next address
+     * generated is outside of the lookahead range
+     */
+    it('should detect ismine up to the lookahead', async ($) => {
+      const info = await wclient.getAccount(watchOnlyWalletId, 'default');
+      await wclient.execute('selectwallet', [watchOnlyWalletId]);
+
+      const xpub = testxpub(0, network);
+      for (let i = 0; i < info.lookAhead; i++) {
+        const address = toAddress(xpub, {
+          network,
+          branch: 0,
+          index: i,
+          enc: 'string'
+        });
+
+        const response = await wclient.execute('getaddressinfo', [address]);
+        assert.equal(response.ismine, true);
+      }
+
+      // derive an address at lookahead+1
+      const failed = toAddress(xpub, {
+        network,
+        branch: 0,
+        index: info.lookahead+1,
+        enc: 'string'
+      });
+
+      const response = await wclient.execute('getaddressinfo', [failed]);
+      assert.equal(response.ismine, false);
+    });
+
+    /*
+     * match bitcoind returning an rpc error when sending the
+     * incorrect network
+     */
+    it('should throw for the wrong network', async () => {
+      const xpub = testxpub(0, network);
+      const failed = toAddress(xpub, {
+        network: 'main',
+        branch: 0,
+        index: 0,
+        enc: 'string'
+      });
+
+      const fn = async () => await wclient.execute('getaddressinfo', [failed]);
+      await assert.asyncThrows(fn, 'Invalid address.');
+    });
+
+    it('should fail for invalid address', async () => {
+      const xpub = testxpub(0, network);
+      let failed = toAddress(xpub, {
+        network,
+        branch: 0,
+        index: 0
+      });
+      // remove the first character
+      failed = failed.slice(1, failed.length);
+
+      const fn = async () => await wclient.execute('getaddressinfo', [failed]);
+      await assert.asyncThrows(fn, 'Invalid address.');
+    });
+  });
+});
+
+/*
+ * helper functions
+ */
+function testxpub(index = 0, network) {
```

Comment:
This may actually not need to be a function because the same arguments are used throughout the test, you can likely create the key at the initial setup for the test.

Reply:
Good point, I removed this function. I see that in one of your tests you hard code in the key from the well known mnemonic. I could derive everything with a script and hard code in all of the keys if that is the style we want to stick with

Reply:
I realized that the mnemonic is defined in `test/data/mnemonic-english.json`, so I am now using that one, along with hard coded addresses derived from that mnemonic